File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    
    bool isItPossible(char s1, char s2) {
          if(s1 == 'z') s1 = 'a';
          else s1 = (char)(int)s1 + 1;
          return s1 == s2;
    }
    
    bool canMakeSubsequence(string str1, string str2) {
        int first = 0;
        
        if(str2.size() > str1.size())return false;
        
        for(int index = 0; index < str1.size(); index++) {
                
               if(str1[index] == str2[first] or isItPossible(str1[index], str2[first])) {
                   first += 1;
                   if(first == str2.size()) {
                       return true;
                   }
               }
        }
        
        return false;
    }
};

File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        i, j = 0, 0
        n = len(str1)
        m = len(str2)
        while i < n and j < m:
            order = ord(str2[j]) - 97
            while i < n and (ord(str1[i]) - 97 + 1) % 26 != order and  str1[i] != str2[j]:
                i += 1
            if i < n:
                i += 1
                j += 1
        return j == m
        